components/bpmn/org.wso2.carbon.bpmn.uuf.ui.stub/src/main/java/org.wso2.carbon.bpmn.uuf.ui/service/BPMNExplorerServiceImpl.java

```diff
@@ -0,0 +1,522 @@
+/*
+ *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.bpmn.uuf.ui;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLContexts;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.nio.charset.Charset;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * Class to call BPMN backend
+ */
+public class BPMNExplorerServiceImpl implements BPMNExplorerService {
+    private static final Logger log = LoggerFactory.getLogger(BPMNExplorerServiceImpl.class);
+
+    private CloseableHttpClient client;
+    private String baseUrl;
+    private String userName;
+    private String passWord;
+
+    /**
+     * Initialize the HttpClient
+     *
+     * @param host
+     * @param port
+     */
+    private void setHttpClient(String host, int port) {
+        try {
+            client = HttpClients.custom()
+                    .setSSLSocketFactory(new SSLConnectionSocketFactory(
+                            SSLContexts.custom().loadTrustMaterial(null, new TrustSelfSignedStrategy()).build()))
+                    .build();
+        } catch (NoSuchAlgorithmException | eyManagementException | KeyStoreException e) {
+            log.error("Exception : ", e);
+        }
+        baseUrl = "https://" + host + ":" + port;
+        //Need to change this part after introducing user managemnet module
+        userName = "admin";
```

Comment:
do not use hard coded user names, passwords, Read those either through the api or using a configuration file